python/cudf/cudf/core/index.py

```diff
@@ -2116,6 +2118,10 @@ def __repr__(self):
         return "\n".join(lines)
 
     def __getitem__(self, index):
+        if type(self) == IntervalIndex:
+            raise NotImplementedError(
+                "Iteration over an IntervalIndex is not yet" " supprted"
```

Comment:
This should be that "getting a scalar from an IntervalIndex is not yet supported". We'll never support iteration.

Reply:
Sounds good!